evalml/pipelines/components/estimators/classifiers/lightgbm_classifier.py

```diff
@@ -189,8 +189,12 @@ def fit(self, X, y=None):
             self
         """
         X = infer_feature_types(X)
-        X_encoded = self._encode_categories(X, fit=True)
-        y_encoded = self._encode_labels(y)
+        # --> why wasn't this here before? is it a problem to add
+        if y is not None:
+            y = infer_feature_types(y)
```

Comment:
we weren't initializing woodwork for y if it wasn't already set, which is problematic for `_handle_nullable_types` which would init woodwork with type inference, which might not be all that different than what ends up happening in `infer_feature_types`, but I figure we should use the common util when we can.